WordSearch.java

```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class WordSearch {
+    private List<String> results = new ArrayList<>();
+
+    public void findAllWords(int col, int row, char[][] characters, Dictionary dictionary) {
+        for (int i = 0; i < characters.length; i++) {
+            for (int j = 0; j < characters[1].length; j++) {

```

Comment:
`characters[1]` can be null (ex. `characters = {{'a'}}`), which would throw a `NullPointerException`. So using `col` and `row` would be a much better approach, I guess.